src/main/java/cms/gov/madie/measure/services/VersionService.java

```diff
@@ -99,16 +111,43 @@ private void validateMeasureForVersioning(Measure measure, String username) {
     }
   }
 
-  protected void getNextVersion(Measure measure, String versionType) {
-    if (VERSION_TYPE_MAJOR.equalsIgnoreCase(versionType)) {
-      measure.getVersion().setMajor(measure.getVersion().getMajor() + 1);
-      measure.getVersion().setMinor(0);
-      measure.getVersion().setRevisionNumber(0);
-    } else if (VERSION_TYPE_MINOR.equalsIgnoreCase(versionType)) {
-      measure.getVersion().setMinor(measure.getVersion().getMinor() + 1);
-      measure.getVersion().setRevisionNumber(0);
-    } else if (VERSION_TYPE_PATCH.equalsIgnoreCase(versionType)) {
-      measure.getVersion().setRevisionNumber(measure.getVersion().getRevisionNumber() + 1);
+  protected Version getNextVersion(Measure measure, String versionType) {
+    Version version;
+    try {
+      if (VERSION_TYPE_MAJOR.equalsIgnoreCase(versionType)) {
+        version =
+            measureRepository
+                .findMaxVersionByMeasureSetId(measure.getMeasureSetId())
+                .orElse(new Version());
+        return version.toBuilder().major(version.getMajor() + 1).minor(0).revisionNumber(0).build();
+
+      } else if (VERSION_TYPE_MINOR.equalsIgnoreCase(versionType)) {
+        version =
+            measureRepository
+                .findMaxMinorVersionByMeasureSetIdAndVersionMajor(
+                    measure.getMeasureSetId(), measure.getVersion().getMajor())
+                .orElse(new Version());
+        return version.toBuilder().minor(version.getMinor() + 1).revisionNumber(0).build();
+
+      } else if (VERSION_TYPE_PATCH.equalsIgnoreCase(versionType)) {
+        version =
+            measureRepository
+                .findMaxRevisionNumberByMeasureSetIdAndVersionMajorAndMinor(
+                    measure.getMeasureSetId(),
+                    measure.getVersion().getMajor(),
+                    measure.getVersion().getMinor())
+                .orElse(new Version());
+        return version.toBuilder().revisionNumber(version.getRevisionNumber() + 1).build();
+      }
+    } catch (RuntimeException ex) {
+      log.error("VersionService::getNextVersion Exception while getting version number", ex);
+      throw new InternalServerErrorException(
+          "Unable to version measure with id: " + measure.getId(), ex);
     }
+    return new Version();
+  }
+
+  private String getLibraryContentLine(String cqlLibraryName, Version version) {
```

Comment:
`get` is a bit confusing to me since it's pulling the result from anywhere, but creating a new String. Suggest `generate`, `create`, etc etc, instead.

Reply:
@jkotanchik-SB Thanks so much Joe, will do!